ui/index.d.ts

```diff
@@ -32,6 +32,32 @@ export declare class AzureTreeDataProvider implements TreeDataProvider<IAzureNod
     public findNode(id: string): Promise<IAzureNode | undefined>;
     public dispose(): void;
 }
+/**
+ * The TestAzureTreeDataProvider is an implementation of the IAzureTreeDataProvider that automatically signs into
+ * an Azure account via service principals with credentials stored on Travis.  Because signing in is an interactive
+ * experience, this circumvents that to do automated testing with Azure resources.
+ */
+export declare class TestAzureTreeDataProvider implements IAzureTreeDataProvider<IAzureNode>, Disposable {
```

Comment:
Can you point to code that uses this?

Reply:
This class isn't used directly by the azureextensionui package, but is leveraged in azure-functions to automate the login experience for the testing suite.

Reply:
If you push them to a branch on GitHub, they will no longer be just on your local machine...

Reply:
Oh, actually I had pushed up the branch here:
https://github.com/Microsoft/vscode-azurefunctions/tree/nturinski/funcTest

But I haven't created a PR yet as it is still incomplete.  I'm focused on some features for our milestones, so it'll probably be a couple of weeks before I can get time to finish it.